### DISCUSSION TITLE
Use of dot()

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [mjhoptics/ray-optics#73] [Q&A] [answered]
https://github.com/mjhoptics/ray-optics/blob/e55a4c199ddc738235813a08c562d18cce310064/src/rayoptics/elem/transform.py#L70

Hi Mike, trying to understand what the dot product is meant to be - dot between matrix/vector and matrix/matrix - is this just matrix multiplication?

**Comment by mjhoptics** [comment, marked as answer]
This is a matrix-matrix multiplication. Numpy has several ways of doing this, I doubt I had a well thought out reason for choosing this, other than it seemed to work.
